src/groupByKey.ts

```diff
@@ -1,7 +1,22 @@
-type GroupsMap<T> = {
-  [key: string]: T[];
+type GroupsMap<Group> = {
+  [key: string]: Group[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<Group>(
+  items: Group[],
+  key: keyof Group,
+): GroupsMap<Group> {
+  const grouppedItems: GroupsMap<Group> = items.reduce((result, current) => {
+    const itemKey = current[key] as string;
+
+    if (result[itemKey]) {
+      result[itemKey] = [...result[itemKey], current];
+    } else {
+      result[itemKey] = [current];
+    }
+
+    return result;
```

Comment:
The initial value for the reduce operation is cast to `GroupsMap<Group>`, but it would be clearer to provide an actual initial value that matches the expected type, such as `{} as GroupsMap<Group>` to `Object.create(null)`. This avoids potential issues with object prototype inheritance.